spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright © Spyder Project Contributors
+#
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+# -----------------------------------------------------------------------------
+
+"""Tests for the Existing Kernel Connection widget."""
+
+# Third party imports
+import pytest
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import QDialogButtonBox
+import time
+
+# Local imports
+from spyder.plugins.ipythonconsole.widgets.kernelconnect import (
+    KernelConnectionDialog
+)
+from spyder.config.main import CONF
+
+
+# =============================================================================
+# Fixtures
+# =============================================================================
+def setup_dialog(qtbot):
+    """Set up kernel connection dialog."""
+    widget = KernelConnectionDialog()
+    qtbot.addWidget(widget)
+
+    return widget
```

Comment:
You want to make sure the dialog is closed afterward; the example I link below includes that. If you were to use this to return a single dialog and not a factory, you could also do it via changing it to a ``yield`` statement and adding ``widget.close()``, etc. after it.

Reply:
I converted the fixture to be a factory like you suggested. And it works without adding the created dialog to QtBot using `qtbot.addWidget(dialog)`, if I'm not adding the widget anymore, should I still close it?

Reply:
I'd have to actually see your code to be sure, but as long as you're opening and using the dialog as a fixture you want to add a finalizer (like the example I linked did) to close it after.